pageserver/src/basebackup.rs

```diff
@@ -287,7 +291,9 @@ where
                 self.timeline.pg_version,
             )?;
 
-        let lazy_slru_download = self.timeline.get_lazy_slru_download() && !self.full_backup;
+        let lazy_slru_download = (self.lazy_slru_download_enabled
+            || self.timeline.get_lazy_slru_download())
+            && !self.full_backup;
```

Comment:
I don't understand this enablement logic. Lazy SLRU downloads can be enabled either via the compute explicitly requesting it or via the tenant config flag on the pageserver side.

Does the compute side logic for on-demand SLRU download require the flag?

In any case, I don't see the point of the tenant config anymore since roll-out is going to be controlled via cplane.

Reply:
Original idea was to enable lazy SLRU download either for the whole PS (page server config) or particular tenant.
But looks like nobody is going to toggle lazy SLRU download for the concrete tenants.
Instead of it we want to enable it for everybody, but do it in "safe" way: be able to switch it for concrete region and easily switch off in case of error.
This was actually main goal of this PR: to simplify enabling/disabling lazy download.

I agree with you that now tenant config is almost useless, even default config. Should I remove `lazy_srlu_download` from it? I am not sure... Or may be change the logic and make it possible to **disable** lazy SLRU download for particular tenant?

Reply:
> Or may be change the logic and make it possible to disable lazy SLRU download for particular tenant?

Maybe this would be useful. We can change tenant configs on the fly without restart.

Reply:
I thought we wanted to allow for _disabling_ lazy SLRU download for a particular tenant (as you said above).

Reply:
If you specify `lazy_slru_download` (either `on` or `off`) then it overrides all other settings.
So it is possible to disable lazy SLRU download through tenant config.

Reply:
Ah right, I see it now.